src/smolagents/memory.py

```diff
@@ -93,7 +94,7 @@ def to_messages(self, summary_mode: bool = False, show_model_input_messages: boo
                     content=[
                         {
                             "type": "text",
-                            "text": "Calling tools:\n" + str([tc.dict() for tc in self.tool_calls]),
+                            "text": "Calling tools:\n" + json.dumps([tc.dict() for tc in self.tool_calls]),
```

Comment:
@aymeric-roucher 

I see that in the code bellow here https://github.com/huggingface/smolagents/pull/962/files#diff-ff15ad77f081712933e506e9c4c618d724925b2a21bf3eb7ec49abeeeaeb49e6R154
There is an attempt to convert single quote to double quotes so that a python dict becomes json serialisable. But why not to do it right here?

Another option is to go with `| toJson` here: https://github.com/huggingface/smolagents/pull/962/files#diff-ebc6d60cf6cf43b13fe12bc6e43e42158553a5cf39c74892c3b3dd4965da9083R282